start

PawelMaliszewski opened the pull request.

PawelMaliszewski committed:
start

PawelMaliszewski committed:
start

PawelMaliszewski committed:
start

slawekludwiczak commented on src/main/resources/templates/update.html:
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html lang="pl" xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+      layout:decorate="~{layout}">
+<head>
+  <title>Aktualizacja transakcji</title>
+</head>
+<body>
+<main layout:fragment="content">
+  <div class="contactForm" th:fragment="contact">
+    <h2 style="margin-top: 25px">Podaj id transakcji, którą chcesz zmodyfikować.</h2>
+    <form  action="/updateById"  method="get">
+
+      <div class="mb-3 w-25">
+        <label for="id" class="form-label">Wpisz id transakcji</label>
+        <input type="number" name="id" step="1" class="form-control" id="id" min="1" placeholder="id" required>
+      </div>
+
+      <button type="submit" class="btn btn-primary">Wyszukaj</button>
+
+    </form>
+  </div>
+</main>
+</body>
+</html>
```
Zamiast tej strony lepiej dodaj odnośniki do edycji w tabelach z listami przycodów, albo wydatków

slawekludwiczak commented on src/main/resources/templates/error.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="pl"
+      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+      layout:decorate="~{layout}">
+<head>
+    <title>Błąd</title>
+</head>
+<body>
+<main>
+    <div layout:fragment="content" style="text-align: justify">
+        <h1 class="confirmation">Operacja zakończona niepowodzeniem sprawdź, czy wprowadziłeś prawidłowe dane.</h1>
+    </div>
+</main>
+</body>
+</html>
```
Podobnie jak z edycją, usuwanie przenieś lepiej do tabeli z przychodami / wydatkami

slawekludwiczak commented on src/main/java/com/temat24zad1/TransactionRepository.java:
```diff
@@ -0,0 +1,114 @@
+@Repository
+public class TransactionRepository {
+
+    Connection connection;
+
+    public TransactionRepository() {
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+            this.connection = DriverManager.getConnection(
+                    "jdbc:mysql://localhost:3306/transactions", "root", "password");
+        } catch (ClassNotFoundException | SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
```
Wykorzystaj Spring Data JPA i obiekt DataSource, jak pokazałem w code review

slawekludwiczak reviewed (changes requested):
code review https://javastartbootcamp.slack.com/archives/C04CJF5PV0U/p1692346846037289

PawelMaliszewski committed:
Aplikacja po 10 kliknięciach przestaje działać, na serwerze sql nie ma żadnych błędów.
Problem prawdopodobnie jest po stronie springa, po zmianie ustawienia:
spring.datasource.hikari.maximum-pool-size=20,
jest trochę lepiej, ale to chyba nie o to chodzi.

slawekludwiczak reviewed (changes requested):
z przerobioną logiką szablony delete.html i update.html są zbędne

PawelMaliszewski committed:
Usunięte szablony

PawelMaliszewski committed:
Zmiana ustawień connection

PawelMaliszewski committed:
Usunięte szablony
